kotlintest-assertions/kotlintest-assertions-arrow/src/main/kotlin/io/kotlintest/assertions/arrow/either/matchers.kt

```diff
@@ -6,14 +6,20 @@ import io.kotlintest.Result
 import io.kotlintest.matchers.beInstanceOf2
 import io.kotlintest.should
 import io.kotlintest.shouldNot
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+
+@UseExperimental(ExperimentalContracts::class)
+fun <T> Either<*, T>.shouldBeRight() {
+  contract {
+    returns() implies (this@shouldBeRight is Either.Right<*>)
```

Comment:
We wanted this to be the generic type, but it's not possible due to contracts not supporting it yet, right?

Reply:
Yes, so once contracts support generics we can just change `Either.Right<*>` to `Either.Right<T>` and it will be generic.